lib/rubygems/commands/advisory_command.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+require 'rubygems/command'
+require 'rubygems/package'
+require 'rubygems/local_remote_options'
+require 'rubygems/version_option'
+require 'rubygems/gemcutter_utilities'
+
+class Gem::Commands::AdvisoryCommand < Gem::Command
+  include Gem::VersionOption
+  include Gem::GemcutterUtilities
+
+  def initialize
+    super 'advisory', 'add advisories to gem versions'
+
+    add_version_option
+    add_platform_option
+  end
+
+  def arguments # :nodoc:
+    "GEM  name of the gem to be marked vulnerable"
+  end
+
+  def description # :nodoc:
+    <<-EOF
+The advisory command allows you to add advisories to a gem version.
+
+An advisory can be any vulnerability in the gem that makes the gem
+either unusable or risky to use. The risk can or cannot be a security issue.
+
+    EOF
+  end
+
+  def usage # :nodoc:
+    "#{program_name} advisory GEM -v VERSION [-p PLATFORM]"
+  end
+
+  def execute
+    version = get_version_from_requirements(options[:version])
+    platform  = get_platform_from_requirements(options)
+
+    if version
+      send_advisory(version, platform)
+    else
+      say "A version argument is required: #{usage}"
+      terminate_interaction
+    end
+  end
+
+  private
+
+  def send_advisory(version, platform)
+    gem_name = get_one_gem_name
+    alert_warning "Once marked, gem advisories cannot be unmarked for that particular version."
+    title = ask "Title:"
+    description = ask "Description:"
+    url = ask "Url:"
+    cve = ask "Cve:"
```

Comment:
capitalize to `CVE:`

Reply:
In future, I'd consider how the warning and confirmation is presented because you'll find users don't really read such things so it'll be missed. From your screenshots, there's quite a bit of text condensed and easily buries the information. One quick thing I can think of is add new line space before/after warning and confirmation so things are grouped by context.

Since there's no way to edit an advisory for now, you may want to alert user and hopefully they check for typos, and ensure they have good descriptions.

Reply:
What happens if I don't put one of the fields like CVE or description or ? Basically what are the failure messages like?